spec-dtslint/operators/withLatestFrom-spec.ts

```diff
@@ -70,35 +101,102 @@ describe('withLatestFrom', () => {
     it('should infer correctly with 2 params', () => {
       const a = of(1, 2, 3);
       const b = of('a', 'b', 'c');
-      const c = of('d', 'e', 'f');
-      const res = a.pipe(withLatestFrom(b, c, (a, b, c) => b + c)); // $ExpectType Observable<string>
+      const c = of(1, 2, 3);
+      const res = a.pipe(withLatestFrom(b, c, (a, b, c) => c)); // $ExpectType Observable<number>
     });
 
     it('should infer correctly with 3 params', () => {
       const a = of(1, 2, 3);
       const b = of('a', 'b', 'c');
-      const c = of('d', 'e', 'f');
+      const c = of(1, 2, 3);
       const d = of('g', 'h', 'i');
-      const ref = a.pipe(withLatestFrom(b, c, d, (a, b, c, d) => b + c)); // $ExpectType Observable<string>
+      const ref = a.pipe(withLatestFrom(b, c, d, (a, b, c) => b + c)); // $ExpectType Observable<string>
     });
 
     it('should infer correctly with 4 params', () => {
       const a = of(1, 2, 3);
       const b = of('a', 'b', 'c');
-      const c = of('d', 'e', 'f');
+      const c = of(1, 2, 3);
       const d = of('g', 'h', 'i');
-      const e = of('j', 'k', 'l');
-      const res = a.pipe(withLatestFrom(b, c, d, e, (a, b, c, d, e) => b + c)); // $ExpectType Observable<string>
+      const e = of(4, 5, 6);
+      const res = a.pipe(withLatestFrom(b, c, d, e, (a, b, c) => b + c)); // $ExpectType Observable<string>
     });
 
     it('should infer correctly with 5 params', () => {
       const a = of(1, 2, 3);
       const b = of('a', 'b', 'c');
-      const c = of('d', 'e', 'f');
+      const c = of(1, 2, 3);
+      const d = of('g', 'h', 'i');
+      const e = of(4, 5, 6);
+      const f = of('m', 'n', 'o');
+      const res = a.pipe(withLatestFrom(b, c, d, e, f, (a, b, c, d, e) => c + e)); // $ExpectType Observable<number>
+    });
+
+    it('should infer correctly with 6 params', () => {
+      const a = of(1, 2, 3);
+      const b = of('a', 'b', 'c');
+      const c = of(1, 2, 3);
+      const d = of('g', 'h', 'i');
+      const e = of(4, 5, 6);
+      const f = of('m', 'n', 'o');
+      const g = of(7, 8, 9);
+      const res = a.pipe(withLatestFrom(b, c, d, e, f, g, (a, b, c, d, e, f) => b + f)); // $ExpectType Observable<string>
+    });
+
+    it('should infer correctly with array parameter', () => {
+      const a = of(1, 2, 3);
+      const b = of('a', 'b', 'c');
+      const c = of(1, 2, 3);
+      const d = of('g', 'h', 'i');
+      const e = of(4, 5, 6);
+      const f = of('m', 'n', 'o');
+      const g = of(7, 8, 9);
+      const res = a.pipe(withLatestFrom([b, c, d, e, f, g], (a, b) => b)); // $ExpectType Observable<string>
```

Comment:
issue: This inference is not correct. AFAICT, there the implementation does not support an array of observables. It always expects sources to be supplied via a rest parameter. It looks to me like the resultant observable would be `Observable<number | Observable<string | number>>` - as the sources in the array would not be flattened by the implementation.

Reply:
You are right. I removed the signature and also the corresponding test.